Gems/RecastNavigation/Code/Tests/MockInterfaces.h

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Component/Entity.h>
+#include <AzCore/Interface/Interface.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzFramework/Entity/EntityDebugDisplayBus.h>
+#include <AzFramework/Physics/PhysicsScene.h>
+#include <AzFramework/Physics/Shape.h>
+#include <AzTest/AzTest.h>
+#include <LmbrCentral/Shape/ShapeComponentBus.h>
+#include <RecastNavigation/RecastNavigationMeshBus.h>
+#include <AzCore/Time/ITime.h>
+#include <AzCore/Component/TransformBus.h>
+
+namespace RecastNavigationTests
+{
+    using namespace AZ;
+    using namespace AZStd;
+    using namespace testing;
+
+    class MockShapeComponent
+        : public AZ::Component
+        , public LmbrCentral::ShapeComponentRequestsBus::Handler
```

Comment:
update: These Mocks should ideally be added to the Gem that owns the types, not here, so they can be used by other people. Would it be possible to move these to the relevant places if they do not already exist? Thanks!

Reply:
That's great thanks, appreciate you moving these, will definitely be useful for others 👍